### DISCUSSION TITLE
Hi, the problem is that I'm using nextjs 13 and I request a post through postman and I get a response in html format. I will be glad if you can help

### DISCUSSION THREAD

**Discussion by rizvan555** [vercel/next.js#55952] [Help]
### Summary

// /app
//  /api
//   /test
//    /route.ts

'use server';

import express from 'express';
import mongoose from 'mongoose';
import cors from 'cors';
import bodyParser from 'body-parser';
import cookieParser from 'cookie-parser';
import multer from 'multer';
import Cars from '@models/Cars';
import CarEmployment from '@models/CarEmployment';
import { connectToDB } from '@utils/database.js';
import { NextApiRequest, NextApiResponse } from 'next';

mongoose.connect(process.env.MONGODB_COMPAS_CARS_URI, {
  useNewUrlParser: true,
  useUnifiedTopology: true,
});

mongoose.createConnection(process.env.MONGODB_COMPAS_CAREMPLOYMENT_URI, {
  useNewUrlParser: true,
  useUnifiedTopology: true,
});

const app = express();
const port = process.env.PORT || 3000;

app.use(express.json());
app.use(cors());
app.use(cookieParser());
app.use(bodyParser.json({ limit: '50mb' }));
app.use(
  bodyParser.urlencoded({
    limit: '50mb',
    extended: true,
    parameterLimit: 50000,
  })
);
app.use((req, res, next) => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
  next();
});

const storage = multer.memoryStorage();
const upload = multer({
  storage,
  limits: {
    fileSize: 50 * 1024 * 1024,
  },
});

//Cars
app.get('/cars', async (req, res) => {
  try {
    await connectToDB();
    const cars = await Cars.find();
    res.status(200).json(cars);
    console.log(cars);
  } catch (error) {
    console.log(error);
  }
});

app.post('/cars', upload.single('car'), async (req, res) => {
  try {
    await connectToDB();
    const newCar = await Cars.create(req.body);
    res.status(200).json(newCar);
    console.log('Success:', newCar);
  } catch (error) {
    console.log(error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
});

//CarEmployment
app.get('/carEmployment', async (req, res) => {
  try {
    await connectToDB();
    const cars = await CarEmployment.find();
    res.send(cars);
  } catch (error) {
    console.log(error);
  }
});

app.post('/carEmployment', async (req, res) => {
  try {
    await connectToDB();
    const newCar = await CarEmployment.create(req.body);
    console.log('New Car Employment:', newCar);
    res.status(200).json(newCar);
  } catch (error) {
    console.log(error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
});

app.listen(port, () => {
  console.log(`Server is running on port ${port}`);
});

### Additional information

```js
My Post: [
    {
        "img": "/images/mercedes-cla1.jpg.jpg",
        "img1": "/images/mercedes-cla2.jpg.jpeg",
        "img2": "/images/mercedes-cla3.jpg.jpeg",
        "img3": "/images/mercedes-cla4.jpg.jpeg",
        "img4": "/images/mercedes-cla5.jpg.jpeg",
        "name": "Mercedes",
        "model": "CLA",
        "fuel": "Diesel",
        "transmission": "Automatik",
        "displacement": 1.968,
        "owner": "1",
        "price": 200000,
        "year": "2019",
        "km": 8000,
        "ps": 250
    }]

My responce: <!DOCTYPE html>
<html lang="en" class="__className_a64ecd h-full scroll-smooth antialiased">

<head>
	<meta charSet="utf-8" />
	<link rel="preload" as="font" href="/_next/static/media/c9a5bc6a7c948fb0-s.p.woff2" crossorigin=""
		type="font/woff2" />
	<link rel="stylesheet" href="/_next/static/css/app/layout.css?v=1695642821534"
		data-precedence="next_static/css/app/layout.css" />
	<meta name="next-size-adjust" />
	<title>Create Next App</title>
	<meta name="description" content="Generated by create next app" />
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	<script src="/_next/static/chunks/polyfills.js" noModule=""></script>
</head>

<body class="flex h-full flex-col">
	<div style="opacity:0;transform:translateY(15px) translateZ(0)">
		<div class="flex items-center justify-between px-10 py-2 bg-black"><a href="/">
				<h1 class="text-white text-3xl tracking-wider font-bold">LuxuryCars</h1>
			</a>
			<div class="flex items-center gap-6">
				<div class="flex items-center gap-7 px-5 py-4 ">
					<div class="">
						<div><a class="text-white tracking-wider leading-5 text-[14px] " href="/">HOME</a></div>
					</div>
					<div class="">
						<div><a class="text-white tracking-wider leading-5 text-[14px] " href="/about">ABOUT</a></div>
					</div>
					<div class="">
						<div><a class="text-white tracking-wider leading-5 text-[14px] bg-[#00de1b] font-bold pt-3 pr-6 pl-6 pb-3 "
								href="/cars">CARS</a></div>
					</div>
					<div class="">
						<div><a class="text-white tracking-wider leading-5 text-[14px] " href="/news">NEWS</a></div>
					</div>
					<div class="">
						<div><a class="text-white tracking-wider leading-5 text-[14px] " href="/contact">CONTACT</a>
						</div>
					</div>
				</div><a class=" text-white" href="/findCar"><svg stroke="currentColor" fill="currentColor"
						stroke-width="0" viewBox="0 0 24 24" height="23" width="23" xmlns="http://www.w3.org/2000/svg">
						<path
							d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z">
						</path>
					</svg></a>
			</div>
		</div>
	</div>
	<main class="grow">
		<div class="flex flex-col mx-auto pt-10 py-8 relative bg-[#f5f5f5]">
			<div class="flex items-center gap-8 mb-20 ">
				<h1 class="md:text-6xl text-4xl mx-auto md:m-0 font-bold px-5">OUR CARS</h1>
				<hr class="md:hr-line" />
			</div>
			<div class="md:flex flex-wrap grid grid-cols-1 gap-10 justify-center mt-2 mb-6 ">
				<button class="border px-6 py-2 bg-primary text-white myButton md:w-[15vw] w-[40vw] mx-auto md:m-0 ">SEE MORE</button>
			</div>
		</div>
	</main>
	<div class="flex justify-center bg-black text-white py-12 gap-8 relative border-t z-50">
		<div class="absolute flex bg-[#00de1b] text-white top-[-4vh] gap-16 px-10 py-4 rounded ">
			<div class="flex items-center gap-1"><svg stroke="currentColor" fill="currentColor" stroke-width="0"
					viewBox="0 0 16 16" height="20" width="20" xmlns="http://www.w3.org/2000/svg">
					<path fill-rule="evenodd"
						d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z">
					</path>
				</svg>
				<p>+994 123567894</p>
			</div>
			<div class="flex items-center gap-1"><svg stroke="currentColor" fill="currentColor" stroke-width="0"
					viewBox="0 0 24 24" height="20" width="20" xmlns="http://www.w3.org/2000/svg">
					<path fill="none" d="M0 0h24v24H0z"></path>
					<path
						d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z">
					</path>
				</svg>
				<p>Luxury-cars@gmail.com</p>
			</div>
			<div class="flex items-center gap-2"><a href="https://www.facebook.com/"><svg stroke="currentColor"
						fill="currentColor" stroke-width="0" version="1.1" viewBox="0 0 16 16" height="20" width="20"
						xmlns="http://www.w3.org/2000/svg">
						<path
							d="M14.5 0h-13c-0.825 0-1.5 0.675-1.5 1.5v13c0 0.825 0.675 1.5 1.5 1.5h6.5v-7h-2v-2h2v-1c0-1.653 1.347-3 3-3h2v2h-2c-0.55 0-1 0.45-1 1v1h3l-0.5 2h-2.5v7h4.5c0.825 0 1.5-0.675 1.5-1.5v-13c0-0.825-0.675-1.5-1.5-1.5z">
						</path>
					</svg></a><a href="https://www.instagram.com/"><svg stroke="currentColor" fill="currentColor"
						stroke-width="0" viewBox="0 0 448 512" height="24" width="24"
						xmlns="http://www.w3.org/2000/svg">
						<path
							d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z">
						</path>
					</svg></a><a href="https://twitter.com/?lang=de"><svg stroke="currentColor" fill="currentColor"
						stroke-width="0" viewBox="0 0 448 512" height="24" width="24"
						xmlns="http://www.w3.org/2000/svg">
						<path
							d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-48.9 158.8c.2 2.8.2 5.7.2 8.5 0 86.7-66 186.6-186.6 186.6-37.2 0-71.7-10.8-100.7-29.4 5.3.6 10.4.8 15.8.8 30.7 0 58.9-10.4 81.4-28-28.8-.6-53-19.5-61.3-45.5 10.1 1.5 19.2 1.5 29.6-1.2-30-6.1-52.5-32.5-52.5-64.4v-.8c8.7 4.9 18.9 7.9 29.6 8.3a65.447 65.447 0 0 1-29.2-54.6c0-12.2 3.2-23.4 8.9-33.1 32.3 39.8 80.8 65.8 135.2 68.6-9.3-44.5 24-80.6 64-80.6 18.9 0 35.9 7.9 47.9 20.7 14.8-2.8 29-8.3 41.6-15.8-4.9 15.2-15.2 28-28.8 36.1 13.2-1.4 26-5.1 37.8-10.2-8.9 13.1-20.1 24.7-32.9 34z">
						</path>
					</svg></a><a href="https://www.linkedin.com/"><svg stroke="currentColor" fill="currentColor"
						stroke-width="0" viewBox="0 0 448 512" height="24" width="24"
						xmlns="http://www.w3.org/2000/svg">
						<path
							d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z">
						</path>
					</svg></a></div>
		</div>
		<div class="w-[20vw] flex flex-col gap-6">
			<h3 class="text-xl font-bold">CONTACT INFO</h3>
			<p>There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration
				in some form, by injected humour</p>
		</div>
		<div class="w-[20vw] flex flex-col gap-6">
			<h3 class="text-xl font-bold">OPEN HOURS</h3>
			<ul>
				<li>Monday 10:00 am to 08:00 pm</li>
				<li>Monday 10:00 am to 08:00 pm</li>
				<li>Monday 10:00 am to 08:00 pm</li>
				<li>Monday 10:00 am to 08:00 pm</li>
				<li>Monday 10:00 am to 08:00 pm</li>
				<li>Monday 10:00 am to 08:00 pm</li>
				<li>OFF</li>
			</ul>
		</div>
		<div class="w-[20vw] flex flex-col gap-6">
			<h3 class="text-xl font-bold">USEFUL LINK</h3>
			<p>There are many variations of passages of Lorem Ipsum available, but the injected</p>
		</div>
		<div class="flex flex-col w-[20vw] gap-4">
			<h3 class="text-xl font-bold">NEWSLETTER</h3>
			<input type="email" placeholder="Enter your Email" class="pl-2 py-2 text-black" value=""/><button class="myButton pl-2 py-2">SUBSCRIBE</button>
		</div>
	</div>
	<script src="/_next/static/chunks/webpack.js" async=""></script>
	<script src="/_next/static/chunks/main-app.js" async=""></script>
</body>

</html>
<script>
	(self.__next_f=self.__next_f||[]).push([0])
</script>
<script>
	self.__next_f.push([1,"1:HL[\"/_next/static/media/c9a5bc6a7c948fb0-s.p.woff2\",{\"as\":\"font\",\"type\":\"font/woff2\"}]\n2:HL[\"/_next/static/css/app/layout.css?v=1695642821534\",{\"as\":\"style\"}]\n0:\"$L3\"\n"])
</script>
<script>
	self.__next_f.push([1,"4:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/app-router.js\",\"chunks\":[\"app-client-internals:static/chunks/app-client-internals.js\"],\"name\":\"\",\"async\":false}\n6:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/error-boundary.js\",\"chunks\":[\"app-client-internals:static/chunks/app-client-internals.js\"],\"name\":\"\",\"async\":false}\n7:I{\"id\":\"(app-client)/./app/components/Header.tsx\",\"chunks\":[\"app/layout:static/chunks/app/layout.js\"],\"name\":\"\",\"async\":false}\n8:I{\"id\":\"(app-client)/"])
</script>
<script>
	self.__next_f.push([1,"./app/components/Footer.tsx\",\"chunks\":[\"app/layout:static/chunks/app/layout.js\"],\"name\":\"\",\"async\":false}\n9:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/layout-router.js\",\"chunks\":[\"app-client-internals:static/chunks/app-client-internals.js\"],\"name\":\"\",\"async\":false}\na:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/render-from-template-context.js\",\"chunks\":[\"app-client-internals:static/chunks/app-client-internals.js\"],\"name\":\"\",\"async\":false}\nb:I{\"id\":\"(app-client)/./app/"])
</script>
<script>
	self.__next_f.push([1,"cars/page.tsx\",\"chunks\":[\"app/cars/page:static/chunks/app/cars/page.js\"],\"name\":\"default\",\"async\":true}\n"])
</script>
<script>
	self.__next_f.push([1,"3:[[[\"$\",\"link\",\"0\",{\"rel\":\"stylesheet\",\"href\":\"/_next/static/css/app/layout.css?v=1695642821534\",\"precedence\":\"next_static/css/app/layout.css\"}]],[\"$\",\"$L4\",null,{\"assetPrefix\":\"\",\"initialCanonicalUrl\":\"/cars\",\"initialTree\":[\"\",{\"children\":[\"cars\",{\"children\":[\"__PAGE__\",{}]}]},\"$undefined\",\"$undefined\",true],\"initialHead\":[\"$L5\",[\"$\",\"meta\",null,{\"name\":\"next-size-adjust\"}]],\"globalErrorComponent\":\"$6\",\"notFound\":[\"$\",\"html\",null,{\"lang\":\"en\",\"className\":\"__className_a64ecd h-full scroll-smooth antialiased\",\"children\":[\"$\",\"body\",null,{\"className\":\"flex h-full flex-col\",\"children\":[[\"$\",\"$L7\",null,{}],[\"$\",\"main\",null,{\"className\":\"grow\",\"children\":[\"$undefined\",[[\"$\",\"title\",null,{\"children\":\"404: This page could not be found.\"}],[\"$\",\"div\",null,{\"style\":{\"fontFamily\":\"system-ui,\\\"Segoe UI\\\",Roboto,Helvetica,Arial,sans-serif,\\\"Apple Color Emoji\\\",\\\"Segoe UI Emoji\\\"\",\"height\":\"100vh\",\"textAlign\":\"center\",\"display\":\"flex\",\"flexDirection\":\"column\",\"alignItems\":\"center\",\"justifyContent\":\"center\"},\"children\":[\"$\",\"div\",null,{\"children\":[[\"$\",\"style\",null,{\"dangerouslySetInnerHTML\":{\"__html\":\"body{color:#000;background:#fff;margin:0}.next-error-h1{border-right:1px solid rgba(0,0,0,.3)}@media (prefers-color-scheme:dark){body{color:#fff;background:#000}.next-error-h1{border-right:1px solid rgba(255,255,255,.3)}}\"}}],[\"$\",\"h1\",null,{\"className\":\"next-error-h1\",\"style\":{\"display\":\"inline-block\",\"margin\":\"0 20px 0 0\",\"padding\":\"0 23px 0 0\",\"fontSize\":24,\"fontWeight\":500,\"verticalAlign\":\"top\",\"lineHeight\":\"49px\"},\"children\":\"404\"}],[\"$\",\"div\",null,{\"style\":{\"display\":\"inline-block\"},\"children\":[\"$\",\"h2\",null,{\"style\":{\"fontSize\":14,\"fontWeight\":400,\"lineHeight\":\"49px\",\"margin\":0},\"children\":\"This page could not be found.\"}]}]]}]}]]]}],[\"$\",\"$L8\",null,{}]]}]}],\"asNotFound\":false,\"children\":[[\"$\",\"html\",null,{\"lang\":\"en\",\"className\":\"__className_a64ecd h-full scroll-smooth antialiased\",\"children\":[\"$\",\"body\",null,{\"className\":\"flex h-full flex-col\",\"children\":[[\"$\",\"$L7\",null,{}],[\"$\",\"main\",null,{\"className\":\"grow\",\"children\":[\"$\",\"$L9\",null,{\"parallelRouterKey\":\"children\",\"segmentPath\":[\"children\"],\"error\":\"$undefined\",\"errorStyles\":\"$undefined\",\"loading\":\"$undefined\",\"loadingStyles\":\"$undefined\",\"hasLoading\":false,\"template\":[\"$\",\"$La\",null,{}],\"templateStyles\":\"$undefined\",\"notFound\":\"$undefined\",\"notFoundStyles\":\"$undefined\",\"asNotFound\":false,\"childProp\":{\"current\":[\"$\",\"$L9\",null,{\"parallelRouterKey\":\"children\",\"segmentPath\":[\"children\",\"cars\",\"children\"],\"error\":\"$undefined\",\"errorStyles\":\"$undefined\",\"loading\":\"$undefined\",\"loadingStyles\":\"$undefined\",\"hasLoading\":false,\"template\":[\"$\",\"$La\",null,{}],\"templateStyles\":\"$undefined\",\"notFound\":\"$undefined\",\"notFoundStyles\":\"$undefined\",\"asNotFound\":false,\"childProp\":{\"current\":[[\"$\",\"$Lb\",null,{\"params\":{},\"searchParams\":{}}],null],\"segment\":\"__PAGE__\"},\"styles\":[]}],\"segment\":\"cars\"},\"styles\":[]}]}],[\"$\",\"$L8\",null,{}]]}]}],null]}]]\n"])
</script>
<script>
	self.__next_f.push([1,"5:[[[\"$\",\"meta\",null,{\"charSet\":\"utf-8\"}],[\"$\",\"title\",null,{\"children\":\"Create Next App\"}],[\"$\",\"meta\",null,{\"name\":\"description\",\"content\":\"Generated by create next app\"}],null,null,null,null,null,null,null,null,[\"$\",\"meta\",null,{\"name\":\"viewport\",\"content\":\"width=device-width, initial-scale=1\"}],null,null,null,null,null,null,null,null,null,null,[]],[null,null,null,null],null,null,[null,null,null,null,null],null,null,null,null,null]\n"])
</script>
```

**Comment by icyJoseph** [comment]
So you have a Route handler files (route.ts), where you are setting up an Express server, and hope to intercept calls there? I don't think that's gonna work. Route handlers expect named methods to be exported, such as `GET`, `POST`, etc...

You need to refactor so that every Express handler you have is at the correct Route Handler path, for example `app/api/test/cars/route.tsx`, and from there export a GET method handler. There you can invoke your DB connection, and setup different middleware.

In short, the issue is that you are trying to setup an express server within an Route handler file. In App Router you can't do that.

**Reply by rizvan555** [reply]
Thank you very much
